substrafl/nodes/simu_nodes.py

```diff
@@ -0,0 +1,130 @@
+import uuid
+from copy import deepcopy
+
+from substratools import Opener
+from substratools.utils import load_interface_from_module
+
+from substrafl.nodes import AggregationNode
+from substrafl.nodes import TestDataNode
+from substrafl.nodes import TrainDataNode
+from substrafl.nodes.references.local_state import LocalStateRef
+
+
+class SimuTrainDataNode(TrainDataNode):
```

Comment:
I am not sure it is possible to use `typing.Protocol` because both objects will not have the same type signatures:

- `TrainDataNode.update_states` returns a tuple with `LocalStateRef` and a `SharedStateRef`
- `SimuTrainDataNode.update_states` returns a tuple with `LocalStateRef` and not a `SharedStateRef` but the real output of the method (completely arbitrary)

So if we just say that `update_states` returns a tuple it's fine, but it's not really possible to enforce a fine-grained type structure

Reply:
If all the signatures are not the same, I'm even more wary of subclassing, since polymorphism can get nightmarish really fast. 
I think I'd prefer having some loose typing at the Protocol level, and having specialisation at the child class level.

Reply:
OK I'll correct it then!